### PR TITLE
[ticket/14875] Add method for untrimmed input to ajax iohandler

### PR DESCRIPTION
**Pull request by marc1706**
Checklist:

- [x] Correct branch: master for new features; 3.2.x, 3.1.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master / 3.2.x](https://area51.phpbb.com/docs/master/coding-guidelines.html), [3.1.x](https://area51.phpbb.com/docs/31x/coding-guidelines.html)
- [x] Commit follows commit message [format](https://wiki.phpbb.com/Git#Commit_Messages)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-14875

Due to the pre-encoded input and the escaping of the input, the
string has to be decoded twice for the password.

PHPBB3-14875

### CODE REVIEW

**Comment by Nicofuma** [comment]
Couldn't we have a `get_raw_input` or something like that?
(by the way, I wonder what happen if we have an UT8 chars in the password...)

File: phpBB/phpbb/install/helper/iohandler/ajax_iohandler.php [hunk 1]
```diff
@@ -90,12 +90,12 @@ class ajax_iohandler extends iohandler_base
...
 	 *
 	 * @param path_helper						$path_helper
-	 * @param \phpbb\request\request_interface	$request	HTTP request interface
+	 * @param \phpbb\request\request	$request	HTTP request interface
 	 * @param \phpbb\template\template			$template	Template engine
 	 * @param router 							$router		Router
 	 * @param string 							$root_path	Path to phpBB's root
 	 */
-	public function __construct(path_helper $path_helper, \phpbb\request\request_interface $request, \phpbb\template\template $template, router $router, $root_path)
+	public function __construct(path_helper $path_helper, \phpbb\request\request $request, \phpbb\template\template $template, router $router, $root_path)
```
Comment (CHItA) [inline phpBB/phpbb/install/helper/iohandler/ajax_iohandler.php]: Why exactly do you need this?

Comment (marc1706) [inline phpBB/phpbb/install/helper/iohandler/ajax_iohandler.php]: The `raw_variable` method is only part of the request class but not of the request_interface. This was a conscious choice as some other variables of request also are not part of the request_interface, mostly ones that are usually not required for a normal page operation (which IMHO also applies to raw input variables).

Comment (Nicofuma) [inline phpBB/phpbb/install/helper/iohandler/ajax_iohandler.php]: personally I think it could be part of the interface

Comment (Nicofuma) [inline phpBB/phpbb/install/helper/iohandler/ajax_iohandler.php]: I don't thinks there is anyone implementing this interface. And actually I don't thinks we should have one but it does not really matter.

Comment (marc1706) [inline phpBB/phpbb/install/helper/iohandler/ajax_iohandler.php]: Ok, moving it to the interface then.

**Pull request merged**